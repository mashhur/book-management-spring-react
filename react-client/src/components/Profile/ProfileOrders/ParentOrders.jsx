import React from "react";
import RatingComponent from "../../Ratings";
import PropTypes from "prop-types";

const ParentOrders = ({ orders, setChildBooks }) => {
	const getBookName = (books) => {
		return books.length == 1
			? books[0].title
			: `${books[0].title} + ${books.length} more `;
	};

	const viewAllClickHandler = (order) => {
		console.log("setting orders child");
		console.log(order);
		setChildBooks(order);
	};
	return (
		<div
			data-test="component-profile-order-parent"
			className=" container ">
			<div
				className="overflow-scroll px-3"
				style={{ height: "80vh" }}>
				{orders.map((order, index) => (
					<div
						data-test="order-item-component"
						key={index}
						className="product">
						<div className="book ">
							<img src={order.books[0].cover} />
							<div>
								<p
									className={`m-0  p-0 status ${
										order.status == "Cancelled"
											? "text-danger"
											: ""
									}`}>
									{order.statusMsg}
								</p>
								<h3 className="m-0 p-0">
									{getBookName(order.books)}
								</h3>
								<p className="author"> â‚¹{order.price}</p>
								<small className="text-muted">
									{order.orderDate}
								</small>
							</div>
						</div>
						{order.books.length == 1 ? (
							<div className="ratings">
								<RatingComponent book={order.books[0]} />
							</div>
						) : (
							<div className="other">
								<button
									data-test="show-more-books-button"
									onClick={() => viewAllClickHandler(order)}
									className=" badge alert-warning p-2  px-3 btn btn-warning rounded-pill badge-pill">
									view {order.books.length} other books
								</button>
							</div>
						)}
					</div>
				))}
			</div>
		</div>
	);
};

ParentOrders.propType = {
	orders: PropTypes.array.isRequired,
	setChildBooks: PropTypes.func.isRequired,
};
export default ParentOrders;
