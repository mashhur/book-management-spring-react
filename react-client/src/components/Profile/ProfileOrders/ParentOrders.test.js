import React from "react";
import { mount, shallow } from "enzyme";
import {
	findByTestAttr,
	checkProp,
} from "../../../tests/testUtils";
import ParentOrders from "./ParentOrders";
import RatingComponent from "../../Ratings";
import { DummyData } from "../../../tests/dummyData";

const singleBookOrder = DummyData.singleBookOrder; // 1

const multipleBooksOrder = DummyData.multipleBooksOrder; //2

const multipleOrders = DummyData.multipleOrders; //2

const defaultProps = {
	orders: [],
	setchildView: () => {},
	setChildBooks: () => {},
};

const setup = (props = {}) => {
	const setUpProps = { ...defaultProps, ...props };
	return shallow(<ParentOrders {...setUpProps} />);
};

describe("Profile orders [parent] component test casses", () => {
	it("renders without any error", async () => {
		const wrapper = setup();
		const component = await findByTestAttr(
			wrapper,
			"component-profile-order-parent"
		);
		expect(component.length).toBe(1);
	});

	describe("when more books are", () => {
		it("available renders show more", async () => {
			const wrapper = setup({
				orders: multipleBooksOrder,
			});
			const component = await findByTestAttr(
				wrapper,
				"show-more-books-button"
			);
			expect(component.length).toBe(1);
		});

		it("not available renders ratings", () => {
			const wrapper = setup({ orders: singleBookOrder });
			expect(wrapper.find(RatingComponent).exists()).toBe(
				true
			);
		});
	});

	it("should render all the available orders", async () => {
		const wrapper = setup({
			orders: multipleOrders,
		});
		const component = await findByTestAttr(
			wrapper,
			"order-item-component"
		);
		expect(component.length).toBe(multipleOrders.length);
	});

	it("doesnot throw warning with expected props", async () => {
		checkProp(ParentOrders, defaultProps);
	});

	it("should match the snapshot", () => {
		const wrapper = setup({
			orders: multipleOrders,
		});
		expect(wrapper).toMatchSnapshot();
	});
});
