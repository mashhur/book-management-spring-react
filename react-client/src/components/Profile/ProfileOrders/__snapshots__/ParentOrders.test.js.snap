// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile orders [parent] component test casses should match the snapshot 1`] = `ShallowWrapper {}`;
